ProductWarehouse.Application/Features/Queries/GetProducts/ProductsQueryValidator.cs

```diff
@@ -0,0 +1,14 @@
+﻿using FluentValidation;
+
+namespace ProductWarehouse.Application.Features.Queries.GetProducts;
+
+public class ProductsQueryValidator: AbstractValidator<ProductsQuery>
+{
+    public ProductsQueryValidator()
+    {
+        RuleFor(r => r.MinPrice).GreaterThanOrEqualTo(0).WithMessage("MinPrice must be a non-negative value.");
```

Comment:
If you wish, you can put the strings in constants in a class with constants. The decision is yours